packages/medusa/src/services/customer.js

```diff
@@ -387,8 +389,9 @@ class CustomerService extends BaseService {
         customer.email = this.validateEmail_(email)
       }
 
-      if (billing_address) {
-        customer.billing_address = this.validateBillingAddress_(billing_address)
+      if ("billing_address_id" in update || "billing_address" in update) {
+        const address = update.billing_address_id || update.billing_address
+        await this.updateBillingAddress_(customer, address, addrRepo)
       }
```

Comment:
`updateBillingAddress_` can take an address object or address id as the second argument. `billing_address_id` takes precedence.